src/main/java/ezschedule/model/event/Date.java

```diff
@@ -2,24 +2,27 @@
 
 import static java.time.temporal.ChronoUnit.DAYS;
 import static java.time.temporal.ChronoUnit.MONTHS;
-import static java.time.temporal.ChronoUnit.YEARS;
 import static java.util.Objects.requireNonNull;
 
 import java.time.LocalDate;
 import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.time.format.ResolverStyle;
 
 import ezschedule.commons.util.AppUtil;
+import ezschedule.model.event.exceptions.InvalidDateException;
 
 /**
  * Represents an Event's date in the scheduler.
  * Guarantees: immutable; is valid as declared in {@link #isValidDate(String)}
  */
 public class Date implements Comparable<Date> {
 
-    public static final String MESSAGE_CONSTRAINTS =
-            "Date should only contain numeric characters, follows the format yyyy-MM-dd, and it should not be blank";
+    public static final String MESSAGE_CONSTRAINTS = "Date should only contain numeric characters, "
+            + "follows the format yyyy-MM-dd (with year from 2000 to 2999, month from 01 to 12, "
+            + "and day from 01 to 31), and it should not be blank";
 
-    public static final String VALIDATION_REGEX = "^\\d{4}-\\d{2}-\\d{2}$";
+    public static final String VALIDATION_REGEX = "(2\\d{3}-(0[1-9]|1[0-2])-(0[1-9]|[12]\\d|3[01]))";
```

Comment:
is there a need to restrict the year to 2999 as what ho yin mentioned?

Reply:
Fixed